base_agent/dialogue_objects/filter_helper.py

```diff
@@ -174,6 +175,23 @@ def interpret_task_filter(interpreter, speaker, filters_d, get_all=False):
     return F
 
 
+def interpret_dance_filter(interpreter, speaker, filters_d, get_all=False):
+    F = MemoryFilter(interpreter.agent.memory)
+    search_data = {}
+    triples = []
+    for t in filters_d.get("triples"):
+        triples.append(t)
+    search_data["base_table"] = "Dances"
+    search_data["triples"] = triples
+    F.append(BasicFilter(interpreter.agent.memory, search_data))
+    # currently spec intersects all comparators TODO?
+    comparator_specs = filters_d.get("comparator")
```

Comment:
nitpicking here since the variable `comparator_specs` is only being really used here for the `None` check, you could replace it with - 
```
for s in (filters_d.get("comparator") or []):
    F.append(interpret_comparator(interpreter, speaker, s, is_condition=False))
```